### PR TITLE
ENH: Make display names a first class citizen

### PR DESCRIPTION
**Pull request by ant0nsc**
Display names for the submitted jobs can be passed in via an argument now, rather than being set in the submission hook

### CODE REVIEW

File: hi-ml-azure/src/health_azure/himl.py [hunk 1]
```diff
@@ -648,6 +635,8 @@ def submit_run(workspace: Workspace,
         the completion of this run (if True).
     :param wait_for_completion_show_output: If wait_for_completion is True this parameter indicates whether to show the
         run output on sys.stdout.
+    :param display_name: The name that will be displayed in the AML UI. If not provided, the unique run ID will be
+        used as the display name.
```
Comment (fepegar) [inline hi-ml-azure/src/health_azure/himl.py:639]: Does this mean we don't get the random AML names e.g. `hungry_beaver`?

Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/himl.py:639]: you're right, i have  not phrased that correctly

Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/himl.py]: fixed

Comment (fepegar) [reply hi-ml-azure/src/health_azure/himl.py]: Phew. I really enjoyed those :D

**Review by fepegar — APPROVED** [review]

**Review by mebristo — APPROVED** [review]

**Pull request merged**